test/contrib/azureblob_test.py

```diff
@@ -0,0 +1,164 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright 2018 Microsoft Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+"""
+Integration tests for azureblob module.
+"""
+
+import os
+import unittest
+
+import luigi
+from luigi.contrib.azureblob import AzureBlobClient, AzureBlobTarget
+
+client = AzureBlobClient(os.environ.get("ACCOUNT_NAME"), os.environ.get("ACCOUNT_KEY"), os.environ.get("SAS_TOKEN"))
+
+
+class AzureBlobClientTest(unittest.TestCase):
+    def setUp(self):
+        self.client = client
+
+    def tearDown(self):
+        pass
+
+    def test_splitfilepath_blob_none(self):
+        container, blob = self.client.splitfilepath("abc")
+        self.assertEquals(container, "abc")
+        self.assertIsNone(blob)
+
+    def test_splitfilepath_blob_toplevel(self):
+        container, blob = self.client.splitfilepath("abc/cde")
+        self.assertEquals(container, "abc")
+        self.assertEquals(blob, "cde")
+
+    def test_splitfilepath_blob_nested(self):
+        container, blob = self.client.splitfilepath("abc/cde/xyz.txt")
+        self.assertEquals(container, "abc")
+        self.assertEquals(blob, "cde/xyz.txt")
+
+    def test_create_delete_container(self):
+        import datetime
+        import hashlib
+        m = hashlib.md5()
+        m.update(datetime.datetime.now().__str__().encode())
+        container_name = m.hexdigest()
+
+        self.assertFalse(self.client.exists(container_name))
+        self.assertTrue(self.client.create_container(container_name))
+        self.assertTrue(self.client.exists(container_name))
+        self.client.delete_container(container_name)
+        self.assertFalse(self.client.exists(container_name))
+
+    def test_upload_copy_move_remove_blob(self):
```

Comment:
Each method should have it's own tests. Why are they all being crammed into this one test?

Reply:
I was trying to use it as a workflow since even if I move each to it's own test, they would still require each other

Reply:
understood